File: py/Actor.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Actor(MovableObject):
    """ generated source for Actor

    """
    ANIM_DURATION = 100

    def applySpeed(self, d):
        raise NotImplementedError()

    def canJump(self):
        raise NotImplementedError()

    def die(self):
        raise NotImplementedError()

    def getCurrentAnim(self):
        raise NotImplementedError()

    def getFacing(self):
        raise NotImplementedError()

    def getHealth(self):
        raise NotImplementedError()

    def getJumpSpeed(self):
        raise NotImplementedError()

    def getMoveSpeed(self):
        raise NotImplementedError()

    def getXSpeed(self):
        raise NotImplementedError()

    def getYSpeed(self):
        raise NotImplementedError()

    def getZone(self):
        raise NotImplementedError()

    def setJump(self, b):
        raise NotImplementedError()

    def jump(self):
        raise NotImplementedError()

    def setFacing(self, d):
        raise NotImplementedError()

    def resetAccelX(self):
        raise NotImplementedError()

File: pulp/src/diediedie/level/actors/Actor.java
/*
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *  
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *      
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
 *  MA 02110-1301, USA.
 */
 
package diediedie.level.actors;
import diediedie.level.LevelObject;
import diediedie.level.Direction;
import diediedie.level.MovableObject;
import pulpcore.image.CoreGraphics;
import pulpcore.sprite.ImageSprite;
import pulpcore.math.Rect;

import java.util.Set;

/*
 * Actor - interface for Players and Enemies.
 */
public interface Actor extends MovableObject
{            
    public static int ANIM_DURATION = 100;
    void applySpeed(Direction d);
    boolean canJump();
    void die();
    ImageSprite getCurrentAnim();
    Direction getFacing();
    int getHealth();    
    float getJumpSpeed();
    float getMoveSpeed();    
    int getXSpeed();
    int getYSpeed();
    Rect getZone(); //returns the Actor's zone on the current map
    void setJump(boolean b);
    void jump();
    void setFacing(Direction d);
    void resetAccelX();
}  